bin/main.rs

```diff
@@ -75,6 +102,47 @@ struct Args {
         env = "WADM_MAX_JOBS"
     )]
     max_jobs: usize,
+
+    /// URL of the Nats Jetstream KV that will be used for storage
+    #[arg(long = "storage-nkv-url", env = "STORAGE_NATSKV_URL")]
+    storage_nkv_url: Option<String>,
+
+    /// Optional prefix for the server tate that will be stored.
+    /// this prefix will be prepended to the bucket name of the lattice
+    /// (ex. <prefix>_lattice_<lattice id>).
+    #[arg(
+        long = "storage-nkv-lattice-bucket-prefix",
+        env = "STORAGE_NATSKV_LATTICE_BUCKET_PREFIX"
+    )]
+    storage_nkv_lattice_bucket_prefix: Option<String>,
+
+    /// (Optional) NATS NKey authentication
+    #[arg(
+        long = "storage-nkv-nats-auth-nkey",
+        env = "STORAGE_NATSKV_NATS_AUTH_NKEY"
+    )]
+    storage_nkv_nats_auth_nkey: Option<String>,
```

Comment:
Nit: It might be easier to have top level nats creds options that get used for _everywhere_ we use nats connections and get overridden by these. I can do this later if you don't want to do this one